src/entities/widget/model/providers/getWidgetById.ts

```diff
@@ -3,10 +3,9 @@ import {Dispatch} from '@reduxjs/toolkit';
 import {fetchWidgetById} from '@/shared/api/dotsight';
 
 import {WidgetId} from '..';
-
 import {upsert} from '../actions';
 
-export const getWidgetById = (id: WidgetId) => async (dispatch: Dispatch) => {
-  const widget = await fetchWidgetById(id);
-  return dispatch(upsert(widget));
-};
+export const getWidgetById = (id: WidgetId) => async (dispatch: Dispatch) =>
```

Comment:
async function with .catch and .then inside? why the mix?

Reply:
forgot this